celery_unique/core.py

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import datetime
+
+UNIQUE_KEY_PREFIX = 'celery_unique'
+
+
+class UniqueTaskMixin(object):
+    abstract = True
+    unique_key = None
+    unique_backend = None
+
+    def apply_async(self, args=(), kwargs={}, task_id=None, **options):
```

Comment:
The defaults for this method signature should be `None` for two reasons:
1. In `celery.Task.apply_async`, they are defined with `None` defaults.
2. `kwargs` is currently defaulting to a mutable type (a `dict`), which can (and will?) cause problems